src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,17 +1,41 @@
 import React from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as filterActions } from '../../features/filter';
+import { Status } from '../../types/Status';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const query = useAppSelector(state => state.filter.query);
+  const status = useAppSelector(state => state.filter.status);
+  const options = ['all', 'active', 'completed'];
```

Comment:
The 'options' array should be moved outside the component to avoid re-creation on every render. Consider declaring it as a constant at the top of the file.